helper/resource/testing_new.go

```diff
@@ -356,6 +360,39 @@ func runNewTest(ctx context.Context, t testing.T, c TestCase, helper *plugintest
 			continue
 		}
 
+		if step.Query {
+			logging.HelperResourceTrace(ctx, "TestStep is Query mode")
+
+			queryConfigRequest := teststep.ConfigurationRequest{
+				Raw: &step.Config,
+			}
+			err := wd.SetQuery(ctx, teststep.Configuration(queryConfigRequest), step.ConfigVariables)
+			if err != nil {
+				t.Fatalf("Step %d/%d error setting query: %s", stepNumber, len(c.Steps), err)
+			}
+
+			err = runProviderCommand(ctx, t, wd, providers, func() error {
+				return wd.Init(ctx)
+			})
+			if err != nil {
+				t.Fatalf("Step %d/%d error running init: %s", stepNumber, len(c.Steps), err)
+			}
+
+			var queryOut []string
+			err = runProviderCommand(ctx, t, wd, providers, func() error {
+				var err error
+				queryOut, err = wd.Query(ctx)
+				return err
+			})
+			if err != nil {
+				fmt.Printf("Step %d/%d Query Output:\n%s\n", stepNumber, len(c.Steps), queryOut)
+				t.Fatalf("Step %d/%d error running query: %s", stepNumber, len(c.Steps), err)
+			}
+
+			fmt.Printf("Step %d/%d Query Output:\n%s\n", stepNumber, len(c.Steps), queryOut)
```

Comment:
This simply echoes the Terraform output. So a test only fails if `terraform query` fails.

In the real world, we would make assertions, call "query checks," ...